### DISCUSSION TITLE
Recovering a license slot from re-installed Apple laptop

### DISCUSSION THREAD

**Discussion by razmatei** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#897] [General]
Hello.

How can I recover/de-activate a license slot if I forgot to do it before re-installing an Apple laptop ? Do you have any license management system where users autonomously manage their license slots out-of-band ?

Regards.

**Comment by MatthiasKerbl** [comment]
> Hello.
> 
> How can I recover/de-activate a license slot if I forgot to do it before re-installing an Apple laptop ? Do you have any license management system where users autonomously manage their license slots out-of-band ?
> 
> Regards.

Hi @razmatei,

In this case, please write our support. You can find our support email address in the purchasing emails of AlDente Pro.

Cheers,
Matthias

**Comment by Strife97** [comment]
Hello @MatthiasKerbl 
I'm also planning to perform a factory reset on my Macbook, but I'm not sure if I can still retain my current license on the same machine.

**Reply by MatthiasKerbl** [reply]
Hi @Strife97,

I would recommend deactivating the license before the reset.

Cheers,
Matthias

**Comment by razmatei** [comment]
Hello. Well ... I think a better idea would be to consider a licensing system which give some control to the user in terms of activating/de-activating the licenses. I can to some degree understand the reasons behind your decision, but in the same time it is a fact that people forget sometimes (or they could have their machine break totally). When that happens, it should be straightforward to "remove/de-activate" without jumping through unnecessary hoops. If that is not possible, it means the licensing system if half-baked in some sense. Remember that this problem would affect exclusively paying/legit users (which is ironic). I also had a situation where AlDente refused to activate because there are "places" where the traffic was blocked/filtered (there are also those circumstances to consider). The licensing system is not ergonomic or rock-solid in these aspects.

**Reply by MatthiasKerbl** [reply]
Hi @razmatei,

Thank you for your constructive feedback and you are completely right. The current implementation of the licensing solution needs to improve. Unfortunately, it is not trivial and takes time to improve it but we are actively looking into this.

Cheers,
Matthias
